src/components/Planner/Planner.tsx

```diff
@@ -1,37 +1,88 @@
 import React, { useEffect, useState } from "react";
-import { Box, Center, Grid, GridItem, Text } from "@chakra-ui/react";
+import {
+  Center,
+  Grid,
+  GridItem,
+  Text,
+  Image,
+  Container
+} from "@chakra-ui/react";
 import { DragDropContext, DropResult } from "react-beautiful-dnd";
 import { Droppable } from "react-beautiful-dnd";
 import { Draggable } from "react-beautiful-dnd";
 import { getRecipe } from "../../utils/fetchData";
-import { SavedRecipe } from "../../utils/types";
+import { PlannerRecipe } from "../../utils/types";
 
 const Planner = () => {
-  const [recipes, setRecipes] = useState<Array<SavedRecipe>>([]);
+  const [columns, setColumns] = useState({
+    column1: [],
+    column2: [],
+    column3: [],
+    column4: [],
+    column5: [],
+    column6: [],
+    column7: [],
+    column8: []
+  });
+
   useEffect(() => {
     getRecipe()
       .then(response => {
-        setRecipes(response.data.recipes);
+        const fetchedRecipes = response.data.recipes;
+        const column1Items = fetchedRecipes.map((recipe: PlannerRecipe) => ({
+          id: recipe._id,
+          name: recipe.recipeName,
+          image: recipe.recipeImage
+        }));
+        setColumns({
+          column1: column1Items,
+          column2: [],
+          column3: [],
+          column4: [],
+          column5: [],
+          column6: [],
+          column7: [],
+          column8: []
+        });
       })
       .catch(error => {
         console.log(error);
       });
   }, []);
 
-  const handleDragEnd = (result: DropResult) => {
+  type ColumnId = keyof typeof columns;
+
+  const onDragEnd = (result: DropResult) => {
     if (!result.destination) return;
-    const newRecipes = Array.from(recipes);
-    const [reorderedItem] = newRecipes.splice(result.source.index, 1);
-    newRecipes.splice(result.destination.index, 0, reorderedItem);
-    setRecipes(newRecipes);
+
+    const sourceColumnId: ColumnId = result.source.droppableId as ColumnId;
+    const destinationColumnId: ColumnId = result.destination
+      .droppableId as ColumnId;
```

Comment:
I would export the type `Id` from `react-beautiful-dnd` ([this](https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/guides/types.md) is where I found that type) and then use that in your columns type instead. Then you shouldn't have to cast here.

So: 

type TColumns = {[index: Id]: Array<whatever_is_in_there_here>};

Reply:
(see comment above about adding this type when you use `useState`

Reply:
Thank you! I was struggling with this type.